source/IdentityServer3.Localization/Internals/LocalizationServiceFactory.cs

```diff
@@ -12,6 +12,11 @@ public static ILocalizationService Create(LocaleOptions options, IDictionary<str
         {
             var locale = options.GetLocale(env);
 
+            if (string.IsNullOrWhiteSpace(locale))
```

Comment:
Absolutely, right. This was  a miss. Thanks! Could you move this logic into the `GetLocale` method?